### PR TITLE
Migration to Quarkus 3

### PR DESCRIPTION
**Pull request by mkouba**
### TODO
- [x] Mention Maven multi-module projects

CC @gsmet

### CODE REVIEW

File: _posts/2023-05-20-quarkus-3-upgrade.adoc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+* Updates the Quarkus version
+* Transforms the source code to use the `jakarta.*` packages
+* Upgrades the Quarkiverse extensions to the versions compatible with Quarkus 3.0
+* Updates the Java EE API dependencies to the Jakarta EE API versions
+* Adjusts some common dependencies (such as Jackson and Hibernate)
+* Transforms the configuration files for some configuration properties (for example `quarkus.kubernetes.host` to `quarkus.kubernetes.ingress.host`)
+* Renames some common service providers (for example `/META-INF/services/javax.ws.rs.ext.Providers` to `/META-INF/services/jakarta.ws.rs.ext.Providers`)
+
+TIP: Versions defined in the `<properties>` tag in a Maven `pom.xml` should be handled automatically. For example, if the project depends on `javax.enterprise:cdi-api` and the version is specified in a property `<cdi-version>2.0</cdi-version>` then both the dependency and the configured value will be aligned correctly.
```
[the remark this answers is not in the document]
Comment (mkouba) [reply _posts/2023-05-20-quarkus-3-upgrade.adoc:47]: I will remove this TIP..

File: _posts/2023-05-20-quarkus-3-upgrade.adoc [hunk 2]
```diff
@@ -0,0 +1,95 @@
+---
+layout: post
+title: 'Migration to Quarkus 3.0 is a breeze'
+date: 2023-05-20
+tags: migration
+synopsis: 'Quarkus 3.0, our new major release, is out. Now is the time to migrate!'
```
Comment (gsmet) [inline _posts/2023-05-20-quarkus-3-upgrade.adoc:6]: `update` might be a bit better?
```suggestion
synopsis: 'Quarkus 3.0, our new major release, is out. Now is the time to update!'
```

File: _posts/2023-05-20-quarkus-3-upgrade.adoc [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+author: mkouba
+---
+
+Quarkus 3 is out and it's about the time to update your application!
```
Comment (gsmet) [inline _posts/2023-05-20-quarkus-3-upgrade.adoc:10]: Suggested an improvement but I'm not sure we want to push this message. The update to Quarkus 3 might not be easy for people using Hibernate ORM so I wouldn't want people to feel pressured to do it.
```suggestion
Quarkus 3 is out and it's about time to update your application!
```

File: _posts/2023-05-20-quarkus-3-upgrade.adoc [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+The team worked hard to deliver a lot of new features and performance improvements.
+But of course, there are also some breaking changes.
+The https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.0[migration guide] describes all the important changes in the core extensions.
+
+Nevertheless, there is also some tedious work that needs to be done.
+Quarkus 3 has updated to the latest Jakarta EE 10 APIs, which include the infamous change of the `javax.\*` to the `jakarta.*` packages.
+If your application makes use of any Jakarta API, and there's a good chance it does (think about JAX-RS, CDI, JPA, etc.), you'll need to modify your code.
+Furthermore, if there's an explicit Jakarta API dependency in the project you'll need to update this dependency as well.
```
Comment (gsmet) [inline _posts/2023-05-20-quarkus-3-upgrade.adoc:18]:
```suggestion
Furthermore, if there's an explicit Jakarta API dependency in the project, you'll need to update this dependency as well.
```

File: _posts/2023-05-20-quarkus-3-upgrade.adoc [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+. Install the https://quarkus.io/guides/cli-tooling[Quarkus CLI] or make sure a 3+ version is installed
+. Run `quarkus update --stream=3.0`
+. Check the changes made by the tool
+. Profit!
+
+== Quarkus Update Tool - The Full Story
+
+The goal of the update tool is to help developers to migrate an application to Quarkus 3 smoothly.
+The update process can be triggered from the Quarkus CLI, or directly via Maven and Gradle commands.
```
Comment (gsmet) [inline _posts/2023-05-20-quarkus-3-upgrade.adoc:33]:
```suggestion
The update process can be triggered from the Quarkus CLI, or directly via Maven or Gradle commands.
```

File: _posts/2023-05-20-quarkus-3-upgrade.adoc [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+* Updates the Quarkus version
+* Transforms the source code to use the `jakarta.*` packages
+* Upgrades the Quarkiverse extensions to the versions compatible with Quarkus 3.0
+* Updates the Java EE API dependencies to the Jakarta EE API versions
+* Adjusts some common dependencies (such as Jackson and Hibernate)
+* Transforms the configuration files for some configuration properties (for example `quarkus.kubernetes.host` to `quarkus.kubernetes.ingress.host`)
+* Renames some common service providers (for example `/META-INF/services/javax.ws.rs.ext.Providers` to `/META-INF/services/jakarta.ws.rs.ext.Providers`)
+
+TIP: Maven multi-module projects with standard layout are also supported if the parent project imports a Quarkus platform BOM.
```
Comment (gsmet) [inline _posts/2023-05-20-quarkus-3-upgrade.adoc:47]:
```suggestion
TIP: Maven multi-module projects with standard layout are also supported if the parent project imports a Quarkus Platform BOM.
```

File: _posts/2023-05-20-quarkus-3-upgrade.adoc [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+[source,role="secondary asciidoc-tabs-target-sync-gradle"]
+.Gradle
+----
+./gradlew -PquarkusPluginVersion=3.0.3.Final quarkusUpdate --stream=3.0
+----
+
+IMPORTANT: If using https://quarkus.io/guides/cli-tooling[Quarkus CLI] then always make sure the latest version is used. The `quarkus version` command should output version 3+.
+
+Keep in mind that there is no dry-run mode.
+In other words, once you trigger the update the changes will be immediately written to the file system.
```
Comment (gsmet) [inline _posts/2023-05-20-quarkus-3-upgrade.adoc:73]:
```suggestion
In other words, once you trigger the update, the changes will be immediately written to the file system.
```

File: _posts/2023-05-20-quarkus-3-upgrade.adoc [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+However, you can leverage the source code management tool to see the transformations applied.
+For example, if using Git then run the update tool and execute the `git diff` command afterwards to see the changes. 
+And if something went wrong then just use `git checkout .` to revert the changes.
+
+TIP: If your application depends on Hibernate ORM the dedicated https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.0:-Hibernate-ORM-5-to-6-migration[Hibernate ORM 5 to 6] migration guide will come in handy.
```
Comment (gsmet) [inline _posts/2023-05-20-quarkus-3-upgrade.adoc:78]:
```suggestion
TIP: If your application depends on Hibernate ORM, the dedicated https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.0:-Hibernate-ORM-5-to-6-migration[Hibernate ORM 5 to 6] migration guide will come in handy.
```

File: _posts/2023-05-20-quarkus-3-upgrade.adoc [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+That's it.
+Your application should be ready now.
+If there's something wrong please ping us on Zulip or file a https://github.com/quarkusio/quarkus/issues[new issue].
```
Comment (gsmet) [inline _posts/2023-05-20-quarkus-3-upgrade.adoc:82]:
```suggestion
If there's something wrong please, ping us on Zulip or file a https://github.com/quarkusio/quarkus/issues[new issue].
```

File: _posts/2023-05-20-quarkus-3-upgrade.adoc [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+== Summary
+
+In this article, we discussed the possibilities of the Quarkus Update Tool that will make your migration to Quarkus 3.0 a breeze.
+Don't hesitate and go for it!
+
```
Comment (gsmet) [inline _posts/2023-05-20-quarkus-3-upgrade.adoc:88]: Maybe add a note that this tool will be used from now on to provide easy updates?

**Review by gsmet — COMMENTED** [review]
Thanks @mkouba , I added a few suggestions.

File: _posts/2023-05-20-quarkus-3-upgrade.adoc (hunk 5 above)
Comment (mkouba) [reply _posts/2023-05-20-quarkus-3-upgrade.adoc:33]: I'm not sure about this one. There are now two `or`s in the sentence :shrug:

Comment (mkouba) [reply _posts/2023-05-20-quarkus-3-upgrade.adoc]: How about the current wording?

File: _posts/2023-05-20-quarkus-3-upgrade.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (gsmet) [reply _posts/2023-05-20-quarkus-3-upgrade.adoc]: This is a specific use case as the version of the new component is handled in the BOM. Typically in the case of the Quarkiverse extensions, running `quarkus update` will update the property.

File: _posts/2023-05-20-quarkus-3-upgrade.adoc (hunk 3 above)
Comment (gsmet) [reply _posts/2023-05-20-quarkus-3-upgrade.adoc]: Have you seen my comment about not pressuring people?

**Review by gsmet — APPROVED** [review]

**Pull request merged**